### ISSUE TITLE
Need focus state for tags that are used as links 

### ISSUE THREAD

**Issue by thefirstartist** [carbon-design-system/carbon-website#3770]
## Detailed description

**Carbon needs to provide clear guidance for the focus state for tags that are used as links** 

In [Tag](https://carbondesignsystem.com/components/tag/usage), the [accessibility guidance](https://carbondesignsystem.com/components/tag/accessibility#accessibility-considerations) says "Tags that are modified as a link to filter content should also follow the link Carbon Component guidance." However, there's no image example or token for the focus state when users are navigating using the keyboard.

Due to a lack of guidance, we observe different implementations across products. In this wetsonx.ai screen, the focus state is the same as the normal state, which violates accessibility requirements.

![image](https://github.com/carbon-design-system/carbon-website/assets/40870771/33b69021-e255-4dbb-8195-d75de9637987)

**Comment by laurenmrice** [comment]
This guidance will be updated in the open issue #3449 once we have the Clickable tag variant implemented.
